distribution/istio-resources/kustomization.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- deny-all-authorizationpolicy.yaml
+- envoy-filter-ingressgateway-settings.yaml
```

Comment:
Maybe it would be an idea to move `envoy-filter-ingressgateway-settings.yaml` and `envoy-filter-proxy-protocol.yaml` to be alongside the `istio-spec.yaml` as those are needed for a working Istio installation. The other resources should indeed be separate from the Istio installation.

Reply:
Right, the deployment does seem to be a bit hit and miss due to the deployment order. The best solution would be to have these filters be part of the IstioOperator spec, but I'm not sure if you can add resources to it. For now, what you implemented is the best way forward I think.

Speaking of deployment order. Argo CD seems to initially stop retrying an application deployment after 5 failed syncs. I'm not sure what triggers it to try again after a certain period of time, but we should probably look into this. I'm also wondering if it might not be better to use an Argo CD ApplicationSet or Project specification rather than the simple App of Apps we are using now. Especially if either of those allows for defining the dependencies between the applications (and thus hopefully solve the deployment order problem).